BCStudentSoftwareDevTeam/celts
# Fix last few mega branch issues.

Resolved all of the leftover comments from the mega branch. 
Did: 
- Changed the past event notification when a user is looking an event that has already passed.
- Added a check for the user before the user diet update.
- Changed the name of the "Bonner Notes" button to "Add Bonner Note".
- Changed how the JS checks the "Note is for Bonner".
- Made "Note is for Bonner" be checked by default if a user clicks "Add Bonner Note".
- Renamed all of the send recommendation routs so they better match their new purpose. 
- Changed the Program url and description from properties to fields.
- Created a SQL script to update the fields when it gets merged into production.
- Made sure that Bonner was not plural in "Bonner Scholars".
- Added the program an event belongs to to the header on the "Track Volunteers" page.

To test:
- Checkout the branch and reset your database
- Run the test suite and make sure all tests pass
- Run the application and make sure all the card titles match their descriptions on the landing page
- Make sure that Bonner is not plural on its card
- Make sure that if you click on the link to view the CELTS website for the programs you are taken to the correct CELTS web page
- Create an event for any of the programs, return to the landing page, find the program you created an event in, verify the button at the bottom of the card is "View Events", click the "View Events" button and verify it takes you to the correct tab on the "Events List" page
- Go to "Course Management" in the Admin section, Make sure there is a course in the "Approved" section, click the "Download File" button to verify it still works as intended, click "Send Recommendation" to verify it also still works as intended
- On "User Profile" make sure you are still able to create profile notes and that if you create a note from the Bonner section "Note for Bonner" is checked when the notes modal is opened
- Also on "User Profile" go to the "Dietary Restrictions" section and verify you are able to update the users dietary restrictions. You want to do this on a normal user (ayisie) and you want to change to an admin (ramsayb2) and a student staff user (neillz) and make sure you are still able to update the basic users (ayisie) dietary restrictions
- Run the SQL script and make sure it works:
1. In the terminal type _mysql -u root -proot_ then type _use celts;_ (this makes it so you are in mysql using the celts database)
2. run the SQL script with: _source /home/{user}/celts/database/program-url-and-description.sql_
- Verify the program on the "Track Volunteers" page is the correct program
